project/recursive_sorting.py

```diff
@@ -36,20 +36,68 @@ def merge_in_place(arr, start, mid, end):
 
     return arr
 
-def merge_sort_in_place(arr, l, r): 
+
+def merge_sort_in_place(arr, l, r):
     # TO-DO
 
     return arr
 
 
-# TO-DO: implement the Quick Sort function below USING RECURSION
-def quick_sort( arr, low, high ):
+# quick_sort, pivot in middle to avoid disturbing mostly sorted lists
+def quick_sort(arr, low, high):
```

Comment:
great work Kam! I see how it starts from arr[0] all the way up to the pivot, and then starts going through arr[pivot] up to arr[len(arr)] to get it to work. I like that you didn't immediately go to choosing a pivot at the end or the beginning. 
however, I think that, regardless how you choose your pivot, it would make your code a lot shorter if you get the pivot out of the way.

no matter what pivot you choose, you can swap it all the way to the end to "get it out of the way." then, you only need to run your indexes through one type of logic all the way up to the pivot.  
check this link out, and CTRL/COMMAND+ F "Illustration of partition" to see the animation. they choose a pivot right at the very end, but you can follow this if you simply remember to take the middle pivot you found and swap it to the end.

Reply:
no link though